### ISSUE TITLE
Alternative mosaic detection models

### ISSUE THREAD

**Issue by ladaapp** [ladaapp/lada#80]
We currently use YOLOv11-{n,s,m}-seg to detect mosaic region in a video frame.

It works mostly okay but still misses some mosaics. There is still some room for improvement by adjusting the training dataset (https://github.com/ladaapp/lada/issues/23, https://github.com/ladaapp/lada/pull/54).

Probably should do these steps first before diving into alternative models but in any case this ticket is about alternative detection models.

Even though we currently use a segmentation model a detection model would also be fine if it works better. For the restoration pipeline we just need to feed cropped mosaic clips. We don't use the segmentation map for restoration itself so could get away with a model predicing just bounding boxes...

I couldn't find anything that comes close the the runtime performance of YOLO and ease of use. The only interesting thing which may be worth trying is training a detection/segmentation models based on DINOv3 https://github.com/facebookresearch/dinov3

---

Related ticket: Alternative mosaic restoration models https://github.com/ladaapp/lada/issues/7

**Comment by ladaapp** [comment]
Found this DINOv2-based object detection implementation [RF-DETR](https://github.com/roboflow/rf-detr). Did a quick evaluation and trained a small mosaic detection model. Training and integration would be relatively straightforward (compared to not using this library/project and building our own). Cannot tell if its better as I didn't train long enough and didn't use the full dataset.

Next steps:
1) My current mosaic detection dataset for NSFW mosaics is fine but it does not contain enough SFW mosaics (faces, heads). The current detection model does not do a good job in skipping SFW mosaics. Pretty sure its just because of a lack of training data on that front (around 90k NSFW mosaic images vs 3k SFW mosaic images...). Plan is to use [deface](https://github.com/ORB-HD/deface) and run it on the Flickr30k dataset. I did an initial evaluation and the accuracy seems to be quite high compared to other project i have checked/used earlier
2) We currently use YOLO11 segmentation models from ultralytics as a base for the mosaic detection model. They have announced that they will publish their new model in October/2025 including pre-trained segmentation and detection models. Let's wait until this new model is out and train a new model. Effort of setting up a new training run / integration /evaluation should be quite low
3) The maintainer Roboflow of RF-DETR has mentioned that they are working on upgrading their model to DINOv3 and will also publish a segmentation model (no estimates when this work will be done AFAIK). As its a new project it is not as polished as YOLO/ultralytics yet as far as I could see after initial limited evaluation of this. So training and integrating it would take some effort

If 2) or 3) takes longer than expected train a YOLO11m segmentation model based on extended  dataset created in 1). This is the same slow/large architecture used for our current v2 detection model. This should then make v2 and v3.2-accurate obsolete.

**Comment by neru2132** [comment]
Sill discovering this repo. So far it looks very cool! Since you mentioned your datasets a couple of times, would there be any point in crowd sourcing additional images to make them more diverse or would you expect that to not be of much help?
